service.go

```diff
@@ -74,40 +75,42 @@ func (svc *Service) StartSubscription(ctx context.Context, sub *nostr.Subscripti
 					}).Errorf("Failed to process event: %v", err)
 				}
 				if resp != nil {
-					status := sub.Relay.Publish(ctx, *resp)
+					status, publishErr := sub.Relay.Publish(ctx, *resp)
 					nostrEvent := NostrEvent{}
 					result := svc.db.Where("nostr_id = ?", event.ID).First(&nostrEvent)
 					if result.Error != nil {
 						svc.Logger.WithFields(logrus.Fields{
 							"eventId":      event.ID,
 							"status":       status,
 							"replyEventId": resp.ID,
+							"publishErr": publishErr,
 						}).Error(result.Error)
 						return
 					}
 					nostrEvent.ReplyId = resp.ID
-					// https://github.com/nbd-wtf/go-nostr/blob/master/relay.go#L321
-					if status == nostr.PublishStatusSucceeded {
-						nostrEvent.State = NOSTR_EVENT_STATE_PUBLISH_CONFIRMED
-						nostrEvent.RepliedAt = time.Now()
+					
+					if status == nostr.PublishStatusFailed || publishErr != nil {
```

Comment:
I wouldn't switch around the cases here without a good reason?

Reply:
now I also check if the error is not nil. I think the error should be checked before proceeding. Maybe there is a better way?

Reply:
Edit: I saw your other comment.